src/go/rpk/pkg/cli/cmd/group/offset_delete.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package group
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+)
+
+func newOffsteDeleteCommand(fs afero.Fs) *cobra.Command {
+	return &cobra.Command{
+		Use:   "offset-delete GROUP FILE",
+		Short: "Delete offsets for a kafka group",
+		Long: `Forcefully delete offsets for a kafka group.
+
+The broker will only allow the request to succeed if the group is in a dead
+state (no subscriptions) or there are no subscriptions for offsets for
+topic/partitions requested to be deleted.
+
+Indicate which offsets are to be deleted by providing the correctly formatted
+text file as the second command line argument. The file must contain lines of topic
+then partition seperated by a tab. Example:
+
+topic_a<tab>0
+topic_a<tab>1
+topic_b<tab>0
+`,
+		Args: cobra.ExactArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			adm, err := kafka.NewAdmin(fs, p, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			group := args[0]
+			topicPartitionsFile := args[1]
+
+			topicsSet, err := parseTopicPartitionsFile(fs, topicPartitionsFile)
+			out.MaybeDie(err, "unable to parse file: %v", err)
+
+			responses, err := adm.DeleteOffsets(context.Background(), group, topicsSet)
+			out.MaybeDie(err, "unable to delete offsets: %v", err)
+
+			tw := out.NewTabWriter()
+			defer tw.Flush()
+			for topic, partitionErrors := range responses {
+				for partition, error := range partitionErrors {
+					msg := "OK"
+					if error != nil {
+						msg = err.Error()
+					}
+					fmt.Fprintf(tw, "%s\t%d\t%s\n", topic, partition, msg)
+				}
+			}
+		},
+	}
+}
+
+func parseTopicPartitionsFile(fs afero.Fs, file string) (kadm.TopicsSet, error) {
```

Comment:
This can be replaced with `out.ParseFileArray`, i.e.,

type offsetDeleteRow {
  topic string
  partition int32
}

out.ParseFileArray[offsetDeleteRow](fs, file)
```